.travis.yml

```diff
@@ -1,31 +1,66 @@
+env:
+  global:
+    # include $HOME/.local/bin for `aws`
+    - PATH=$HOME/.local/bin:$PATH
 language: shell
 notifications:
   email:
     on_success: never
 stages:
-- tests
+- unit_tests
+- build
+- e2e_tests
 - deployment
+before_install:
+  # set up awscli packages
+  - pip install --user awscli
+  # set up shared storage for build artifacts
+  - mkdir -p ~/build-stages-shared-storage/$TRAVIS_BUILD_NUMBER
+  - aws s3 sync s3://aws-log-forwader-travis-build-shared-storage/$TRAVIS_BUILD_NUMBER ~/build-stages-shared-storage/$TRAVIS_BUILD_NUMBER
 jobs:
   include:
-    - stage: tests
-      name: Tests & Linting
+    - stage: unit_tests
+      name: Execute unit tests & linting
       language: shell
       services: docker
       script:
         - docker build -f pipeline/Dockerfile -t lambda_image .
         - docker run --rm lambda_image bash -c 'python3 -m pylint --rcfile=pipeline/pylint.cfg $(find src/ -name "*.py")'
         - docker run --rm lambda_image bash -c 'cd src && python3 -m pytest -v ../tests/unit'
+    - stage: build
+      name: Build AWS Lambda package
+      #if: tag =~ /^release.*$/
+      language: shell
+      services: docker
+      script:
+        - docker run --rm -v `pwd`:/var/task amazon/aws-sam-cli-build-image-python3.8 bash -c './build-release-package.sh'
+        - cp -p dynatrace-aws-log-forwarder.zip ~/build-stages-shared-storage/$TRAVIS_BUILD_NUMBER
+        - cp -p dynatrace-aws-log-forwarder-lambda.zip ~/build-stages-shared-storage/$TRAVIS_BUILD_NUMBER
+    - stage: e2e_tests
+      name: Execute e2e tests
+      # run e2e tests only for development branches until test execution part is done

```

Comment:
why? I see value in running deployment test without final test ready

Reply:
There is value in running deployment without actually testing the log forwarder functionality. My motivation here was to save resources and build time until the entire suite is completed. 

Point taken. I will update.

Reply:
Code updated.